include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
+ * This function processes the following events:
+ * - Incoming messages from the server <br>
+ * Topic filter subscriptions registered with @ref mqtt_function_subscribe instruct
+ * the server to send matching messages to this client. These messages are received
+ * by this function, which then calls the appropriate message handler. This function
+ * must be called at a rate faster than the incoming messages, as it is the only way
+ * the client receives processing time to manage incoming messages.
+ * - MQTT keep-alive (sending ping requests and processing ping responses) <br>
+ * The MQTT keep-alive mechanism involves sending pings to the server if the connection
+ * is idle. Therefore, in the absence of any other messages, <b>this function must be called
+ * at least once every keep-alive period to send the ping request</b>.
+ * - @ref mqtt_autoreconnect (if enabled) <br>
+ * If the client detects a disconnect, the reconnection will be performed in this function.
  *
- * @param pClient Reference to the IoT Client
- * @param timeout_ms Maximum number of milliseconds to pass thread execution to the client.
+ * @param[in] pClient MQTT client context
+ * @param[in] timeout_ms Amount of time to yield. This function will return to the caller
+ * after AT LEAST this amount of thime has passed.
  *
- * @return An IoT Error Type defining successful/failed client processing.
- *         If this call results in an error it is likely the MQTT connection has dropped.
- *         iot_is_mqtt_connected can be called to confirm.
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
+ * @return If this call results a negative value, it is likely the MQTT connection has dropped.
+ * @ref mqtt_function_is_client_connected can be called to confirm.
  */
+/* @[declare_mqtt_yield] */
 IoT_Error_t aws_iot_mqtt_yield(AWS_IoT_Client *pClient, uint32_t timeout_ms);
+/* @[declare_mqtt_yield] */
 
 /**
- * @brief MQTT Manual Re-Connection Function
+ * @brief Attempt to reconnect with the MQTT server.
+ *
+ * This function makes a single reconnect attempt with the server. If the
+ * reconnection is successful, subscriptions from the client's previous
+ * session are restored as well.
+ *
+ * If this function fails, the client's state is set to `CLIENT_STATE_PENDING_RECONNECT`.
  *
- * Called to establish an MQTT connection with the AWS IoT Service
- * using parameters from the last time a connection was attempted
- * Use after disconnect to start the reconnect process manually
- * Makes only one reconnect attempt Sets the client state to
- * pending reconnect in case of failure
+ * @param[in] pClient MQTT client context
  *
- * @param pClient Reference to the IoT Client
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  *
- * @return An IoT Error Type defining successful/failed connection
+ * @note Generally, it is not necessary to call this function if @ref mqtt_autoreconnect
+ * is enabled. Nevertheless, this function may still be called to initiate a reconnect.
```

Comment:
How can the application know whether the re-connect was successful to avoid calling this?

Reply:
Yield will return SUCCESS.  When auto reconnect is enabled, reconnects are handled transparently by yield and the application never needs to call this function.

Reply:
Okay, then can you instead mention that this function can be called if the auto-reconnect attempts through the `yield` call(s) are exhausted?

Reply:
Added note